app/main/views/agreements.py

```diff
@@ -0,0 +1,24 @@
+from flask import render_template
+from flask_login import login_required
+
+from .. import main
+from ..auth import role_required
+from ... import data_api_client
+from . import get_template_data
+
+
+@main.route('/agreements/<framework_slug>', methods=['GET'])
+@login_required
+@role_required('admin-ccs-sourcing')
```

Comment:
This should probably be `role_required('admin', 'admin-ccs-sourcing')`, unless we have a reason to disallow this for admin.